application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -55,7 +55,8 @@
     private static final ScheduledExecutorService ROUTINE_SERVICE =
             Executors.newScheduledThreadPool(5);
     private final Config config;
-    private final Map<String, SlashCommand> nameToSlashCommands;
+    private final Map<String, UserInteractor> nameToInteractor;
```

Comment:
Seems really vulnerable, if we add context-commands with the same name as a slash-command this won't work

Reply:
each slash command is also a userinteractor at the same time. so `nameToSlashCommands` is a subset of `nameToInteractor`. At first I hoped I can get away with just one of the maps, but then I needed access to slash commands specifically.

I can probably still get around it by doing `(SlashCommand) nameToInteractor.get(name)` and that way only have one map.

Reply:
Issue is that there will be multiple commands with the same name in the future

Reply:
Why that? That shouldnt exist. That would mess with the system and as a solution we would need to add prefixes like "slashcommand;foo;..." to differentiate it from "whatevercommand;foo;...". But why do we need to support that?

Reply:
So in case we want to have 2 commands with the same name, you want us to just name them differently? Why shouldn't it exist? It makes sense for the user's UX, 2 commands with the same behaviour, with the same name.
The system isn't made for it indeed, I have to create an issue for.

Reply:
You mean like being able to trigger, for example `/ban` also via user context menu? I see, makes sense. Yes, the system is not prepared for that yet. It is not very difficult though, prefixes will do the trick.